src/App.tsx

```diff
@@ -1,14 +1,54 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+  const [selectedItem, setSelectedItem] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos().then(result => {
+      setTodos(result);
+      setLoading(false);
```

Comment:
What happens if promise finishes with negative result (server error, disconnected, etc)? what will happen to your loading state? (hint, consider to add catch handler here)